src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorHelpers.cs

```diff
@@ -394,67 +410,120 @@ int sgn(T val)
+
+							if (pixelShouldBeBlank)
+							{
+								bgraCheckeredPixelData.Add(0);
+								bgraCheckeredPixelData.Add(0);
+								bgraCheckeredPixelData.Add(0);
+								bgraCheckeredPixelData.Add(0);
+							}
+							else
+							{
+								bgraCheckeredPixelData.Add((byte)(checkerColor.B * checkerColor.A / 255));
+								bgraCheckeredPixelData.Add((byte)(checkerColor.G * checkerColor.A / 255));
+								bgraCheckeredPixelData.Add((byte)(checkerColor.R * checkerColor.A / 255));
+								bgraCheckeredPixelData.Add(checkerColor.A);
+							}
 						}
 					}
-				}
-			});
-
-			//if (asyncActionToAssign != null)
-			//{
-			//	asyncActionToAssign.Cancel();
-			//}
+				};
 
-			//asyncActionToAssign = ThreadPool.RunAsync(workItemHandler);
-			//asyncActionToAssign.Completed = new AsyncActionCompletedHandler(
-			//async (IAsyncAction asyncInfo, AsyncStatus asyncStatus) =>
-			//{
-			//	if (asyncStatus != AsyncStatus.Completed)
-			//	{
-			//		return;
-			//	}
-
-			//	asyncActionToAssign = null;
+			if (asyncActionToAssign is { })
+			{
+				asyncActionToAssign.Cancel();
+			}
 
-			// Uno Doc: Assumed normal priority is acceptable
-			await dispatcherHelper.RunAsync(CoreDispatcherPriority.Normal, () =>
+			// Uno Doc: ThreadPool is not implemented, so we work around it
+
+			// asyncActionToAssign = ThreadPool.RunAsync(workItemHandler);
+			// asyncActionToAssign.Completed(winrt::AsyncActionCompletedHandler(
+			// 	[width, height, bgraCheckeredPixelData, &asyncActionToAssign, completedFunction, dispatcherQueue]
+			// 	(winrt::IAsyncAction asyncInfo, winrt::AsyncStatus asyncStatus)
+			// {
+			// 	if (asyncStatus != winrt::AsyncStatus::Completed)
+			// 	{
+			// 		return;
+			// 	}
+			//
+			// 	asyncActionToAssign = nullptr;
+			//
+			// 	dispatcherQueue.TryEnqueue(winrt::DispatcherQueueHandler([completedFunction, width, height, bgraCheckeredPixelData]()
+			// 	{
+			// 		winrt::WriteableBitmap checkeredBackgroundBitmap = CreateBitmapFromPixelData(width, height, bgraCheckeredPixelData);
+			// 		completedFunction(checkeredBackgroundBitmap);
+			// 	}));
+			// }));
+
+			await Task.Run(() => workItemHandler(new UIAsyncOperation(null)));
+
+			await EnqueueAsync(dispatcherQueue, () =>
 			{
 				WriteableBitmap checkeredBackgroundBitmap = CreateBitmapFromPixelData(width, height, bgraCheckeredPixelData);
 				completedFunction?.Invoke(checkeredBackgroundBitmap);
 			});
-			//});
+		}
+
+		// Copied from WindowsCommunityToolkit/Microsoft.Toolkit.Uwp/Extensions/DispatcherQueueExtensions.cs
+		// Licensed to the .NET Foundation under one or more agreements.
+		// The .NET Foundation licenses this file to you under the MIT license.
+		// See the LICENSE file in the project root for more information.
+		private static Task EnqueueAsync(this DispatcherQueue dispatcher, Action function, DispatcherQueuePriority priority = DispatcherQueuePriority.Normal)
```

Comment:
So this is brand-new and doesn't exist in WinUI it seems. Not sure that is the right approach then. I didn't need to do this in the original port and just using async/await and Task was fine. Calculating a bitmap in a byte array can be done on any thread.

Reply:
This is takes from the community toolkit and seems like a very reasonable extension of `TryEnqueue` that should probably have been present iniaitially. This is different from the WinUI source because C++ doesn't have async/await. so `TryEnqueue` works for everything. Probably fine anyway, but as usual I try to make the code as close to the original.

Reply:
Same comments as above. When functionality simply doesn't exist my instinct is to go with C#/.NET convention rather than try to shoe-horn in a **new** solution somewhere in-between C#/.NET and C++ all in an attempt to make the code "closer to the original". It seems better not to do new things here.

Someone else is going to have to make the final decision here.